components/common/context/themeContext.tsx

```diff
@@ -0,0 +1,81 @@
+import path from 'path';
+import React, { createContext, useContext, useState, useEffect } from 'react';
+import fs from 'fs';
+
+type ColorValues = {
+  text_color: string;
+  node_color: string;
+  headers_color: string;
+  background_color: string;
+  child_active_connection_color: string;
+  parent_active_connection_color: string;
+  connection_color: string;
+};
+
+type LangValues = {
+  rtl: boolean;
+};
+
+type ThemeContextValues = {
+  themeColors?: ColorValues;
+  lang: LangValues;
+  translationsMap: Map<string, any>; // Add translationsMap to context
+  fetchTranslations: () => Promise<void>; // Method to fetch translations
+};
+
+const themeContext = createContext<ThemeContextValues | undefined>(undefined);
+
+export function useThemeContext() {
+  const context = useContext(themeContext);
+  if (context === undefined) {
+    throw new Error('useThemeContext must be used within a ThemeProvider');
+  }
+  return context;
+}
+
+export function ThemeProvider({ children, rootContent, lang }: {
+  children: React.ReactNode;
+  rootContent: { data: Array<{ Key: string; Value: string }> };
+  lang: string;
+}) {
+  const colorValues: ColorValues = rootContent.data.reduce((acc, curr) => {
+    acc[curr.Key as keyof ColorValues] = curr.Value;
+    return acc;
+  }, {} as ColorValues);
+
+  const [rtl, setRtl] = useState(lang === 'AR');
+  const [translationsMap, setTranslationsMap] = useState<Map<string, any>>(new Map());
+  const fetchTranslations = async () => {
+    let translations: Record<string, unknown> = {};
+    try {
+      const response = await fetch(`/assets/locales/${lang}.json`);
+      if (!response.ok) {
+        throw new Error('Network response was not ok');
+      }
+      translations = await response.json();
+    } catch (error) {
+      console.error('Error fetching translations:', error);
+    }
+    const newTranslationsMap = new Map<string, any>();
+    Object.entries(translations).forEach(([key, value]) => {
+      newTranslationsMap.set(key, value);
+    });
+
+    setTranslationsMap(newTranslationsMap);
+  };
```

Comment:
Are you sure this part of the code is not called in the Frontend? if it's not called then it's okay. But if it's called then we should move it to server-side props and no need to have fetchTranslations function, the context itself will need those translations to be passed to it as props, like we are passing themeColors.

You can double-check that by:
- console.log inside the function to see if that is printed on the browser console
- adding a delay (sleep) getting the files to see what is happening exactly in the FE

`const sleep = ms => new Promise(r => setTimeout(r, ms));`